File: F# Workspace/01_Basics/01_Basics/circle_calc.fs
(*
 * circle_calc.fs
 * By Antonio F. Huertas
 * This program calculates and displays the area and circumference of the
 * circle with the given radius.
 *)

// pi : float
// Defines the value of pi.
let pi = 3.14159

// area_circle : float -> float
// Returns the area of the circle with the given radius.
let area_circle rad = pi * rad ** 2.0

// circum_circle : float -> float
// Returns the circumference of the circle with the given radius. 
let circum_circle rad = 2.0 * pi * rad

// start : float -> unit
// Serves as an entry point for the program.
let start rad =
    printfn "The radius of the circle is %.3f" rad
    if rad < 0.0 then
        printfn "Error! Negative radius."
    else
        printfn "The area is %.3f" (area_circle rad)
        printfn "The circumference is %.3f" (circum_circle rad)

File: Prolog Workspace/01_Basics/circle_calc.pl
/*
 * circle_calc.pl
 * By Antonio F. Huertas
 * This program displays the area and circumference of a circle.
 */

%% pi(?ValuePi)
%  Defines the value of pi.
pi(3.14159).

%% area_circle(+Radius, ?Area)
%  Computes the area of the circle with the given radius.
area_circle(Radius, Area) :-
    pi(ValuePi),
    Area is ValuePi * Radius ** 2.0.

%% circumference_circle(+Radius, ?Circum)
%  Computes the circumference of the circle with the given radius.
circumference_circle(Radius, Circum) :-
    pi(ValuePi),
    Circum is 2 * ValuePi * Radius.

%% start(+Radius)
%  Serves as an entry point for the program.
start(Radius) :-
    write('The radius is '), write(Radius), nl,
    Radius >= 0,

    area_circle(Radius, Area),
    write('The area is '), write(Area), nl,

    circumference_circle(Radius, Circum),
    write('The circumference is '), write(Circum), nl.